packages/flutter_tools/test/integration.shard/deprecated_gradle_settings_test.dart

```diff
@@ -16,6 +17,40 @@ void main() {
     final String workingDirectory = fileSystem.path.join(getFlutterRoot(), 'dev', 'integration_tests', 'gradle_deprecated_settings');
     final String flutterBin = fileSystem.path.join(getFlutterRoot(), 'bin', 'flutter');
 
+    final File settingsDotGradleFile = fileSystem.file(
+        fileSystem.path.join(workingDirectory, 'android', 'settings.gradle'));
+    const String settingsDotGradle = r"""
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// This is the `settings.gradle` file that apps currently use.
+// This file has changed, so it must be migrated in existing projects.
+
+include ':app'
+
+def flutterProjectRoot = rootProject.projectDir.parentFile.toPath()
+def localPropertiesFile = new File(rootProject.projectDir, "local.properties")
+def properties = new Properties()
+
+def plugins = new Properties()
+def pluginsFile = new File(flutterProjectRoot.toFile(), '.flutter-plugins')
+if (pluginsFile.exists()) {
+    pluginsFile.withReader('UTF-8') { reader -> plugins.load(reader) }
+}
+
+plugins.each { name, path ->
+    def pluginDirectory = flutterProjectRoot.resolve(path).resolve('android').toFile()
+    include ":$name"
+    project(":$name").projectDir = pluginDirectory
+}
+""";
+
+    expect(
+      settingsDotGradleFile.readAsStringSync().trim().replaceAll('\r', ''),
```

Comment:
Nit: settingsDotGradleFile and settingsDotGradle are very similar and it took a bit of back and forth for me to understand the expected difference. Can you name settingsDotGradle to instead be prefixed with expected and moved the contents of the string to a referenced const.

Reply:
You made the change I was asking for but then all the code referenced here was removed. I think that is ok because packages/flutter_tools/test/integration.shard/android_plugin_skip_unsupported_test.dart covers the test conditions but with the move plus test rename I cant tell for sure. 

If you think packages/flutter_tools/test/integration.shard/android_plugin_skip_unsupported_test.dart covers the right test conditions then this is resolved.

Reply:
Yes `android_plugin_skip_unsupported_test` should cover the tests from the expected `settings.gradle`.
#140037 is just the basic test against the settings.gradle without the special testing for unsupported android plugins, which was not directly related to this PR.